compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1122,12 +1156,14 @@ fn should_encode_const(def_kind: DefKind) -> bool {
 
 impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
     fn encode_attrs(&mut self, def_id: LocalDefId) {
-        let mut attrs = self
-            .tcx
+        let tcx = self.tcx;
+        let mut is_public: Option<bool> = None;
```

Comment:
Why is the handling of `privacy_access_levels` so complicated?
What about caching it once in `EncodeContext` and then unconditionally call `.get(&def_id)` on it?
r=me once perf lands